### PR TITLE
Added tolerations for Strawberry

### PR DESCRIPTION
**Pull request by qurname2**
To resolve https://github.com/ytsaurus/ytsaurus-k8s-operator/issues/327

### CODE REVIEW

File: pkg/components/microservice.go [hunk 1]
```diff
@@ -129,6 +133,7 @@ func (m *microserviceImpl) rebuildDeployment() *appsv1.Deployment {
 			Image: m.image,
 		},
 	}
+	m.builtDeployment.Spec.Template.Spec.Tolerations = m.tolerations
```
Comment (l0kix2) [inline pkg/components/microservice.go:136]: Aren't they already being set in `deployment.Build()`?

File: pkg/components/strawberry_controller.go [hunk 2]
```diff
@@ -221,6 +226,7 @@ func (c *StrawberryController) syncComponents(ctx context.Context) (err error) {
 			VolumeMounts: volumeMounts,
 		},
 	}
+	deployment.Spec.Template.Spec.Tolerations = c.microservice.getTolerations()
```
Comment (l0kix2) [inline pkg/components/strawberry_controller.go:229]: Aren't they being set in `microservice.buildDeployment()` ?

File: pkg/components/microservice.go (hunk 1 above)
Comment (qurname2) [reply pkg/components/microservice.go:136]: Yes, looks like you right, overengineered that a bit =)

File: pkg/components/strawberry_controller.go (hunk 2 above)
Comment (qurname2) [reply pkg/components/strawberry_controller.go:229]: Yes, looks like you right, overengineered that a bit =)

File: pkg/components/microservice.go [hunk 3]
```diff
@@ -46,6 +46,7 @@ type microserviceImpl struct {
 	builtDeployment *appsv1.Deployment
 	builtService    *corev1.Service
 	builtConfig     *corev1.ConfigMap
+	tolerations     []corev1.Toleration
```
Comment (l0kix2) [inline pkg/components/microservice.go:49]: It looks like you don't need it in this struct

File: pkg/components/microservice.go [hunk 4]
```diff
@@ -54,12 +55,14 @@ func newMicroservice(
...
+	tolerations []corev1.Toleration) microservice {
 	return &microserviceImpl{
 		labeller:      labeller,
 		image:         image,
 		instanceCount: instanceCount,
 		ytsaurus:      ytsaurus,
+		tolerations:   tolerations,
 		service: resources.NewHTTPService(
 			serviceName,
 			nil,
```
Comment (l0kix2) [inline pkg/components/microservice.go:68]: but need to pass tolerations in NewDeployment instead.

**Review by l0kix2 — CHANGES_REQUESTED** [review]

File: pkg/components/strawberry_controller.go [hunk 5]
```diff
@@ -64,6 +65,9 @@ func NewStrawberryController(
 		if resource.Spec.StrawberryController.Image != nil {
 			image = *resource.Spec.StrawberryController.Image
 		}
+		if len(resource.Spec.StrawberryController.Tolerations) != 0 {
+			tolerations = resource.Spec.StrawberryController.Tolerations
+		}
```
Comment (l0kix2) [inline pkg/components/strawberry_controller.go:70]: No point of this check, just pass `resource.Spec.StrawberryController.Tolerations` in newMicroservice

File: pkg/components/microservice.go (hunk 4 above)
Comment (qurname2) [reply pkg/components/microservice.go]: fixed

File: pkg/components/microservice.go (hunk 3 above)
Comment (qurname2) [reply pkg/components/microservice.go]: fixed

**Review by l0kix2 — APPROVED** [review]

**Pull request merged**